Add tabbed examples for multiple node IDs (#2512)
neo4j/docs-operations: modules/ROOT/pages/import.adoc

== Using multiple node IDs

A node header can contain multiple `ID` columns.

> Starting from 2025.07, the relationship data must then use a matching number of `START_ID` / `END_ID` columns as references to the composite value of those ID columns.

Reviewer: While you're right that it is still allowed to use a single :START_ID/:END_ID to refer to a composite ID, we don't want to document that behavior necessarily. It was never really meant to work that way and it is only still possible to not break potential usages of this behavior.

We want to encourage using a matching number of `START_ID` / `END_ID` columns and the other possibility was not meant to exist and should therefor IMO not be documented.

This implies using `string` as `id-type`.

For each `ID` column, you can specify to store its values as different node properties.
However, the composite value cannot be stored as a node property.

[IMPORTANT]
====
Incremental import doesn't support the use of multiple node identifiers.
This functionality is only available with a full import.
====

=== Define multiple IDs as node properties

. Define multiple `ID` columns in the node header.
+
.nodes_header.csv
[source, csv]
----
:ID,:ID,name
----
+
.nodes.csv
[source, csv]
----
aa,11,John
bb,22,Paul
----

. Define the relationship between two established nodes.
+
[.tabbed-example]
=====

> [role=include-with-multiple-node-IDs]

Author: I don't like these headings. In fact, the first tab is on how to use single :START_ID/:END_ID column, and the second one is for matching number of those columns.
Suggestions:
The first tab -- Using single ID columns for composite IDs
The second tab -- Using multiple ID columns for composite IDs

======

Now use both IDs when defining the relationship:

.relationships_header.csv
[source, csv]
----
:START_ID,:TYPE,:END_ID
----

.relationships.csv
[source, csv]
----
aa11,WORKS_WITH,bb22
----

======
[role=include-with-multiple-ID-columns label--new-2025.07]
======

> Starting from 2025.07, you have to use a matching number of `START_ID` / `END_ID` columns when defining the relationship:

Author: ```suggestion
Starting from 2025.07, you can use a matching number of `START_ID` / `END_ID` columns when defining the relationship.
However, do not mix how to refer to composite IDs.
Either all references must use a single `START_ID` / `END_ID` column, or a matching number of them.
```

----
aa,11,WORKS_WITH,bb,22
----
======
=====

[[multiple-IDs-Id-spaces]]
=== Define multiple IDs stored in ID spaces

. Define a `MyGroup` ID space in the _nodes_header.csv_ file.
+

> [source, csv]

Author: ```suggestion
.nodes_header.csv
[source, csv]
```

----
personId:ID(MyGroup),memberId:ID(MyGroup),name
----
+

> [source, csv]

Author: ```suggestion
.nodes.csv
[source, csv]
```
